### PR TITLE
Update `slcli firewall detail` to handle multi vlan firewalls

### PR DESCRIPTION
**Pull request by BrianSantivanez**
Issue: https://github.com/softlayer/softlayer-python/issues/1650
Observations: Command was updated to handle multi vlan firewalls and unit tests was added 

image
image

### CODE REVIEW

File: SoftLayer/CLI/firewall/detail.py [hunk 1]
```diff
@@ -12,31 +12,67 @@
...
+            table.add_row(['public vlan', utils.lookup(_firewall, 'networkGateway', 'publicVlan', 'vlanNumber')])
+            table.add_row(['private vlan', utils.lookup(_firewall, 'networkGateway', 'privateVlan', 'vlanNumber')])
+            table.add_row(['type', _firewall.get('firewallType')])
+
+            if credentials:
+                table.add_row(['FortiGate username', utils.lookup(_firewall, 'managementCredentials', 'username')])
+                table.add_row(['FortiGate password', utils.lookup(_firewall, 'managementCredentials', 'password')])
+
+            rules = mgr.get_dedicated_fwl_rules(firewall_id)
+            table.add_row(['rules', get_rules_table(rules)])
```
Comment (caberos) [inline SoftLayer/CLI/firewall/detail.py:64]: please add a conditional to rules is empty

File: SoftLayer/CLI/firewall/detail.py [hunk 2]
```diff
@@ -12,31 +12,67 @@
...
+
+        if firewall_type == 'multiVlan':
+            _firewall = mgr.get_instance(firewall_id)
+
+            table = formatting.KeyValueTable(['name', 'value'])
+            table.align['name'] = 'r'
+            table.align['value'] = 'l'
+
```
Comment (caberos) [inline SoftLayer/CLI/firewall/detail.py:49]: uppercase or lowercase, please use the same rule on the names of table

**Review by caberos — CHANGES_REQUESTED** [review]

File: SoftLayer/CLI/firewall/detail.py (hunk 1 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/firewall/detail.py:64]: A conditional to print rules was added where rules variable is empty

File: SoftLayer/CLI/firewall/detail.py (hunk 2 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/firewall/detail.py:49]: the names of of table were changed to lowercase

File: SoftLayer/CLI/firewall/detail.py [hunk 3]
```diff
@@ -12,31 +12,70 @@
...
 
     mgr = SoftLayer.FirewallManager(env.client)
 
     firewall_type, firewall_id = firewall.parse_id(identifier)
-    _firewall = mgr.get_instance(firewall_id)
 
-    table = formatting.KeyValueTable(['name', 'value'])
-    table.align['name'] = 'r'
-    table.align['value'] = 'l'
+    if firewall_type == 'vs' or firewall_type == 'server' or firewall_type == 'vlan' or firewall_type == 'multiVlan':
```
Comment (allmightyspiff) [inline SoftLayer/CLI/firewall/detail.py:25]: This is a little easier to work with.
```suggestion
    if firewall_type in ('vs', 'server', 'vlan', 'multiVlan'):
```

File: SoftLayer/CLI/firewall/detail.py [hunk 4]
```diff
@@ -12,31 +12,70 @@
 
 @click.command(cls=SoftLayer.CLI.command.SLCommand, )
 @click.argument('identifier')
+@click.option('--credentials', type=click.BOOL,
+              help="Display FortiGate username and FortiGate password to multi vlans.")
 @environment.pass_env
-def cli(env, identifier):
+def cli(env, identifier, credentials):
     """Detail firewall."""
```
Comment (allmightyspiff) [inline SoftLayer/CLI/firewall/detail.py:19]: I think this command would benefit from an example in this docblock.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Looks good, just a few minor changes needed. Thanks.

File: SoftLayer/CLI/firewall/detail.py (hunk 3 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/firewall/detail.py:25]: The conditional was updated following the suggestion

File: SoftLayer/CLI/firewall/detail.py (hunk 4 above)
Comment (BrianSantivanez) [reply SoftLayer/CLI/firewall/detail.py]: Two examples about vs and multi vlan were added in dockblock

**Review by caberos — APPROVED** [review]
LGTM

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**